File: PinewoodDerbyIot/app/src/main/java/com/johnanderson/pinewoodderbyiot/MyApplication.java
package com.johnanderson.pinewoodderbyiot;

import com.johnanderson.pinewoodderbyiot.di.AppComponent;
import com.johnanderson.pinewoodderbyiot.di.AppInjector;
import com.johnanderson.pinewoodderbyiot.di.DaggerAppComponent;

import dagger.android.AndroidInjector;
import dagger.android.support.DaggerApplication;
import timber.log.Timber;

public class MyApplication extends DaggerApplication {

    private AppComponent appComponent;

    public MyApplication() {
        super();
        appComponent = DaggerAppComponent.builder().application(this).build();
    }

    @Override
    protected AndroidInjector<? extends DaggerApplication> applicationInjector() {
        return appComponent;
    }

    @Override
    public void onCreate() {
        super.onCreate();
        Timber.plant(new Timber.DebugTree());
        AppInjector.INSTANCE.init(appComponent, this);
    }
}

File: PinewoodDerbyIot/app/src/main/java/com/johnanderson/pinewoodderbyiot/MyApplication.kt
package com.johnanderson.pinewoodderbyiot

import android.app.Application
import timber.log.Timber.DebugTree
import timber.log.Timber



/**
 * Created by johnanderson on 12/10/17.
 */
class MyApplication: Application() {
    override fun onCreate() {
        super.onCreate()
        Timber.plant(DebugTree())
    }
}